File: .claude/skills/jimmer-orm/assets/java/EntityTemplate.java
package com.example.model;

import org.babyfish.jimmer.sql.*;
import org.jetbrains.annotations.Nullable;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

/**
 * 实体模板 - 包含常见字段和关联类型
 */
@Entity
public interface EntityTemplate {

    // ========== 主键 ==========
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long id();

    // ========== 业务键 ==========
    @Key
    String name();

    // ========== 基础字段 ==========
    String description();

    @Nullable
    String optionalField();

    BigDecimal amount();

    int count();

    boolean active();

    LocalDateTime createdTime();

    @Nullable
    LocalDateTime updatedTime();

    // ========== 多对一关联 ==========
    @ManyToOne
    @Nullable
    ParentEntity parent();

    // ========== 一对多关联（镜像） ==========
    @OneToMany(mappedBy = "parent")
    List<ChildEntity> children();

    // ========== 多对多关联 ==========
    @ManyToMany
    @JoinTable(
        name = "ENTITY_RELATED_MAPPING",
        joinColumnName = "ENTITY_ID",
        inverseJoinColumnName = "RELATED_ID"
    )
    List<RelatedEntity> relatedEntities();
}

File: .claude/skills/jimmer-orm/assets/kotlin/EntityTemplate.kt
package com.example.model

import org.babyfish.jimmer.sql.*
import java.math.BigDecimal
import java.time.LocalDateTime

/**
 * 实体模板 - 包含常见字段和关联类型
 */
@Entity
interface EntityTemplate {

    // ========== 主键 ==========
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long

    // ========== 业务键 ==========
    @Key
    val name: String

    // ========== 基础字段 ==========
    val description: String

    val optionalField: String?

    val amount: BigDecimal

    val count: Int

    val active: Boolean

    val createdTime: LocalDateTime

    val updatedTime: LocalDateTime?

    // ========== 多对一关联 ==========
    @ManyToOne
    val parent: ParentEntity?

    // ========== 一对多关联（镜像） ==========
    @OneToMany(mappedBy = "parent")
    val children: List<ChildEntity>

    // ========== 多对多关联 ==========
    @ManyToMany
    @JoinTable(
        name = "ENTITY_RELATED_MAPPING",
        joinColumnName = "ENTITY_ID",
        inverseJoinColumnName = "RELATED_ID"
    )
    val relatedEntities: List<RelatedEntity>
}
